### PR TITLE
Fixes #17230 - Adds tracer support

### PR DESCRIPTION
**Pull request by sean797**

### CODE REVIEW

**Comment by sean797** [comment]
Requires https://github.com/Katello/katello-agent/pull/38

File: db/migrate/20161031204903_add_tracer_support.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class AddTracerSupport < ActiveRecord::Migration
+  def change
+    create_table :katello_host_tracers do |t|
+      t.references 'host', :null => false, :index => true
+      t.string 'application'
+      t.string 'helper'
+      t.string 'app_type'
```
Comment (jlsherrill) [inline db/migrate/20161031204903_add_tracer_support.rb]: would add :null => false to the appropriate entries here to match the validations

File: app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb [hunk 2]
```diff
@@ -138,6 +138,15 @@ def upload_package_profile
       render :json => Resources::Candlepin::Consumer.get(@host.subscription_facet.uuid)
     end
 
+    api :PUT, "/consumers/:id/tracer", N_("Update services requiring restart")
+    param :traces, Hash, :required => true
+    def upload_tracer_profile
+      User.as_anonymous_admin do
+        async_task(::Actions::Katello::Host::UploadTracerProfile, @host, params[:traces])
```
Comment (jlsherrill) [inline app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb]: I imagine these tracer uploads are going to be fairly small?  You could probably get away with doing this outside of a dynflow action by just doing the `host.import_tracer_profile(profile)` within this controller.  It doesn't cause any communication with candlepin or pulp, and I wouldn't expect it to be a long process.

**Comment by jlsherrill** [comment]
Looks like its failing the bastion linting:

```
00:11:04.686 app/assets/javascripts/bastion_katello/content-hosts/content/content-host-traces.controller.js
00:11:04.686   43:8  error  Expected indentation of 8 characters  indent
```

Overall looks quite good :+1: will test it out shortly

**Comment by sean797** [comment]
Made those changes, I've also added a few more tests and added `scoped_search` so traces are searchable :+1:

**Comment by jlsherrill** [comment]
ACK thanks @sean797 !

**Pull request merged**